client/state/data-layer/wpcom/sites/hosting/test/restore-database-password.js

```diff
@@ -0,0 +1,50 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+
+/**
+ * Internal Dependencies
+ */
+import {
+	requestRestoreDatabasePassword,
+	showSuccessNotification,
+	showErrorNotification,
+} from '../restore-database-password';
+import { HOSTING_RESTORE_DATABASE_PASSWORD } from 'state/action-types';
+import { errorNotice, successNotice } from 'state/notices/actions';
+
+describe( 'requestRestoreDatabasePassword', () => {
+	it( 'should return an http action with the proper path', () => {
```

Comment:
I have my doubts about the value of this sort of test, they can tend to be a lot of boilerplate to just double check implementation details, ie. seems a bit like doing: 

```javascript
const myObject = { name: 'bob }; 
expect( myObject.name ).toEqual( myObject.name );
```
The likes of the reducer tests where you are verifying the logic of data transformations, etc. make perfect sense however. 

Just my personal opinion though - feel free to disregard - and now the tests are in place I don't see any issue with merging and just revisiting if they become a pain point in future refactoring.

Reply:
> I have my doubts about the value of this sort of test, they can tend to be a lot of boilerplate to just double check implementation details

You're right, and this is actually included in our [unit testing guidelines](https://wpcalypso.wordpress.com/devdocs/docs/testing/unit-tests.md):

> As with all code, tests have to be maintained. Writing tests for the sake of having a test isn't the goal – rather we should try to strike the right balance between covering expected and unexpected behaviours, speedy execution and code maintenance.
>
> Unit tests test units of behavior, and should contain relatively few abstractions. If our tests are failing in response to legitimate changes, it could be a sign that we have coupled our tests too closely to the internals of our code.

Will push a change removing these tests.

Reply:
Done in 25861f1.

I also removed some tests I added in #37966 that were checking implementation details rather than behavior.

I left the selector test there because I think it's important to ensure that the selector returns the values assumed by the components using it:
- `null` when data has not been requested.
- list of users when data has been requested.